libraries/AP_Motors/AP_MotorsMatrix.cpp

```diff
@@ -302,8 +302,12 @@ void AP_MotorsMatrix::output_armed_stabilizing()
         }
     }
 
+    // determine throttle thrust for harmonic notch
+    const float throttle_thrust_best_plus_adj = throttle_thrust_best_rpy + thr_adj;
```

Comment:
I guess the change to what is being passed into the check_for_failed_motor not related to the the dynamic filter?
@lthall, you're aware of this change of course?

I'm a little worried that _throttle_out is only being set for the AP_MotorsMatrix but AC_AttitudeControl_Multi::get_notch_freq_scaled() uses it for all multicopters.  So for example TriCopters will have _throttle_out = 0 and so their get_notch_freq_scaled() will return zero I think.

I guess we are completely sure that "compensation-gain" will never be zero?  I think we are but perhaps a sanity check is worth while so we don't get a divide-by-zero and then have horror unleashed.

Reply:
This is @lthall 's change and design, slightly modified by me to account for all the changes that have gone into 4.0. I defer to him on whether it is right or not.

Reply:
Ok I have sanity checked the compensation gain. get_compensation_gain() already has checks, but I guess it can't hurt

Reply:
If compensation_gain is zero we are already crashing because all control is set to zero.